# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/cocoa/nsDragService.h
# Commit: 384b025f56d3
# Full Hash: 384b025f56d3f87e045ecede6b1a16b71c0b5d4f
# Author: David P <daparks@mozilla.com>
# Date: 2024-07-04 16:09:52
# Regressor Bug: 1893119
# File Overlap Count: 1
# Description:
#   Bug 1893119: Part 21 - Separate nsIDragService and nsIDragSession implementations r=gstoll,geckoview-reviewers,rkraesig,win-reviewers,m_kato
#   
#   Split the class inheritance trees for nsIDragService and nsIDragSession.
#   Remember that, before the start of this patch series, the inheritance
#   diagram was:
# ==============================================================================

diff -r beeda5ef3712 -r 384b025f56d3 widget/cocoa/nsDragService.h
--- a/widget/cocoa/nsDragService.h	Thu Jul 04 07:48:11 2024 +0000
+++ b/widget/cocoa/nsDragService.h	Thu Jul 04 07:48:11 2024 +0000
@@ -11,10 +11,10 @@
 
 #include <Cocoa/Cocoa.h>
 
-// Temporary inheritance from nsBaseDragService instead of nsBaseDragSession
-// (which nsBaseDragService temporarily inherits).
-// This will be undone at the end of this patch series.
-class nsDragSession : public nsBaseDragService {
+/**
+ * Cocoa native nsIDragSession implementation
+ */
+class nsDragSession : public nsBaseDragSession {
  public:
   // nsIDragSession
   NS_IMETHOD GetData(nsITransferable* aTransferable,
@@ -34,6 +34,12 @@
       bool aDoneDrag, uint32_t aKeyModifiers) override;
 
  protected:
+  // nsBaseDragSession
+  MOZ_CAN_RUN_SCRIPT virtual nsresult InvokeDragSessionImpl(
+      nsIWidget* aWidget, nsIArray* anArrayTransferables,
+      const mozilla::Maybe<mozilla::CSSIntRegion>& aRegion,
+      uint32_t aActionType) override;
+
   // Creates and returns the drag image for a drag. aImagePoint will be set to
   // the origin of the drag relative to mNativeDragView.
   NSImage* ConstructDragImage(
@@ -59,21 +65,12 @@
   bool mDragImageChanged = false;
 };
 
-// Temporary inheritance from nsDragSession instead of nsBaseDragService
-// (which nsDragSession temporarily inherits).
-// This will be undone at the end of this patch series.
-class nsDragService final : public nsDragSession {
+/**
+ * Cocoa native nsIDragService implementation
+ */
+class nsDragService final : public nsBaseDragService {
  public:
-  nsDragService();
-
-  // nsBaseDragService
-  MOZ_CAN_RUN_SCRIPT virtual nsresult InvokeDragSessionImpl(
-      nsIWidget* aWidget, nsIArray* anArrayTransferables,
-      const mozilla::Maybe<mozilla::CSSIntRegion>& aRegion,
-      uint32_t aActionType) override;
-
- protected:
-  virtual ~nsDragService();
+  already_AddRefed<nsIDragSession> CreateDragSession() override;
 };
 
 #endif  // nsDragService_h_